rewiringamerica/embed.rewiringamerica.org
# New state-specific calculator design

## Description

This implements most of the spec'd design. There are some significant
pieces still missing. They're tracked in Asana, but just as a summary:

- Multiple selection of projects.

- Adding authority logos to the bottom of the page.

- Handling the case of entering a location outside RI.

- Adding an optional email field for people to join the mailing list.

In addition, there are some more pieces missing before this frontend
is ready to replace the existing IRA calculator wholesale:

- Spanish localization

- Analytics

- Appearance customizability via CSS

### Guide to new code

- New .html page for the RI calculator, which pretty much just has the
  embed. It's included in the Parcel targets, which means it will be
  built and deployed. It's not ready for the public yet, so it uses a
  meta tag to disable search engine indexing.

- `state-calculator.ts` is the custom HTML element. This contains all
  the state management.

- `state-incentive-details.ts` is the view layer.

  - A couple of sub-parts are factored out: `utility-selector.ts` and
    `icon-tab-bar.ts`.

- `calculator-types-v1.ts` corresponds to the shape of the API
  responses. This is manually maintained right now, not generated
  from the API in any way; eventually it'd be good to change that.

- `projects.ts` has icons and labels for the various projects.

## Test Plan

- Enter a RI zip code. 02807 for Block Island, 02861 for Providence
  (RI Energy only), 02814 for Pascoag (RIE and PUD available).

- Fill in the rest of the form, click Calculate. Look at results. In
  the 02814 case, switch to the other utility and make sure the
  incentives reload.

- Click the pill buttons to switch projects.

- Click the "Learn more" / "Visit site" buttons.

Resize the window to look at the three possible layouts. (1, 2, or 3
columns of incentive cards.)
